src/team_lib/mhba/xccl_mhba_ib.c

```diff
@@ -6,6 +6,312 @@
 #include "xccl_mhba_ib.h"
 #include "utils/utils.h"
 
+static pthread_spinlock_t mmio_spinlock;
+
+static __attribute__((constructor)) void lock_constructor(void)
+{
+    pthread_spin_init(&mmio_spinlock, PTHREAD_PROCESS_PRIVATE);
+}
+
+static inline void mmio_wc_spinlock(pthread_spinlock_t *lock)
+{
+    pthread_spin_lock(lock);
+    //#if !defined(__i386__) && !defined(__x86_64__)
+    //	/* For x86 the serialization within the spin lock is enough to
+    //	 * strongly order WC and other memory types. */
+    //	mmio_wc_start();
+    //#endif
+}
+
+static inline void mmio_wc_spinunlock(pthread_spinlock_t *lock)
+{
+    /* It is possible that on x86 the atomic in the lock is strong enough
+	 * to force-flush the WC buffers quickly, and this SFENCE can be
+	 * omitted too. */
+    mmio_flush_writes();
+    pthread_spin_unlock(lock);
+}
+
+static void pthread_mmio_write64_be(void *addr, __be64 val)
+{
+    __be32 first_dword  = htobe32(be64toh(val) >> 32);
+    __be32 second_dword = htobe32(be64toh(val));
+
+    /* The WC spinlock, by definition, provides global ordering for all UC
+	   and WC stores within the critical region. */
+    mmio_wc_spinlock(&mmio_spinlock);
+
+    mmio_write32_be(addr, first_dword);
+    mmio_write32_be(addr + 4, second_dword);
+
+    mmio_wc_spinunlock(&mmio_spinlock);
+}
+
+#define HAVE_FUNC_ATTRIBUTE_IFUNC 1
+
+#if HAVE_FUNC_ATTRIBUTE_IFUNC
+void mmio_write64_be(void *addr, __be64 val)
+        __attribute__((ifunc("resolve_mmio_write64_be")));
+static write64_fn_t resolve_mmio_write64_be(void);
+#else
+__asm__(".type mmio_write64_be, %gnu_indirect_function");
+write64_fn_t resolve_mmio_write64_be(void) __asm__("mmio_write64_be");
+#endif
+
+write64_fn_t resolve_mmio_write64_be(void)
+{
+    return &pthread_mmio_write64_be;
+}
+
+xccl_status_t
+xccl_mhba_ibv_qp_to_mlx5dv_qp(struct ibv_qp *umr_qp, struct internal_qp *mqp)
+{
+    struct mlx5dv_obj dv_obj = {};
```

Comment:
plz remove empty initializer ={} - it is not C standard (will give your warning with -pedantic). you memset the struct to 0 next line anyways.